psi4/src/psi4/libmints/wavefunction.h

```diff
@@ -378,10 +374,22 @@ class PSI_API Wavefunction : public std::enable_shared_from_this<Wavefunction> {
     int get_print() const { return print_; }
     static void initialize_singletons();
 
-    /// Returns the DOCC per irrep array.
-    const Dimension& doccpi() const { return doccpi_; }
-    /// Returns the SOCC per irrep array.
-    const Dimension& soccpi() const { return soccpi_; }
+    /// Returns the DOCC per irrep array. Not recommended for unrestricted code.
+    const Dimension doccpi() const {
+        std::vector<int> docc_vec;
+        for (int h = 0; h < nalphapi_.n(); h++) {
+            docc_vec.push_back(std::min(nalphapi_[h], nbetapi_[h]));
+        }
+        return docc_vec;
+    }
+    /// Returns the SOCC per irrep array. Not recommended for unrestricted code.
+    const Dimension soccpi() const {
+        std::vector<int> socc_vec;
+        for (int h = 0; h < nalphapi_.n(); h++) {
+            socc_vec.push_back(std::abs(nalphapi_[h] - nbetapi_[h]));
+        }
+        return socc_vec;
+    }
```

Comment:
Should this code issue a warning when problematic cases are encountered? In that case, results of any code using socc and docc are incorrect...

Reply:
Problematic meaning `nbeta > nalpha` for at least one irrep, or something else?

Reply:
> Problematic meaning `nbeta > nalpha` for at least one irrep, or something else?

Exactly

Reply:
@loriab Sound like a good candidate for the log file?

Reply:
It'll be the first warning for the log file, but sounds appropriate. Could also repeat in out file if you want.